datasafe-encryption/datasafe-encryption-impl/src/main/java/de/adorsys/datasafe/encrypiton/impl/cmsencryption/CMSEncryptionServiceImpl.java

```diff
@@ -94,6 +122,28 @@ public InputStream buildDecryptionInputStream(InputStream inputStream, Function<
         }
     }
 
+    @Override
+    @SneakyThrows
+    public InputStream buildDecryptionInputStream(InputStream inputStream, Key key, PublicKeyIDWithX509Cert x509Cert) {
```

Comment:
This function should't exist - you don't know key before you open CMS envelope, it must be same lambda-function as is in buildDecryptionInputStream, but accepting list of key-names

Reply:
done